server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -244,6 +249,7 @@ public VS toValuesSource(QueryShardContext context) {
             } else if (valueSourceType() == ValuesSourceType.ANY || valueSourceType() == ValuesSourceType.BYTES) {
                 vs = (VS) ValuesSource.Bytes.WithOrdinals.EMPTY;
             } else {
+                // TODO: Do we need a missing case for Range values type?
                 throw new IllegalArgumentException("Can't deal with unmapped ValuesSource type " + valueSourceType());
```

Comment:
So this is actually the case where the field is unmapped _and_ the user has specified a [`missing` parameter](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-histogram-aggregation.html#_missing_value_13) to fill in missing values with some kind of default.

Assuming we support the `missing` param on ranges too, I think we'll need to set a valid VS here.  We'll also need something down below that does the equivalent of `MissingValues.replaceMissing()` with the user-supplied range.

That said... I'm not entirely sure how or if this works with ranges.  The `missing` param in other aggs only deals with a string or numeric, not a complex to/from object.  So perhaps we just don't support `missing`, but in that case we'll have to make sure it's [not added to the parser](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceParserHelper.java#L68-L69) for ranges.  I'm not opposed to skipping support for `missing` on first iteration and adding it later if requested by users.  Being a runtime feature it won't have any BWC implications and we can add later if desired.


(The unmapped (but no `missing`) scenario is handled above where vs is set to `null`, which should be fine.)